python/jupyter_leaflet/src/controls/GeomanDrawControl.ts

```diff
@@ -525,4 +539,47 @@ export class LeafletGeomanDrawControlView extends LeafletControlView {
     }
     this.layers_to_data();
   }
+
+  model_events() {
+    super.model_events();
+    // Geoman needs to be forced to update by removing and re-adding the control
+    // toolbar with the new options set. Ignore attrs that are not options.
+    const excluded_keys = ['current_mode'];
+    for (let key in this.model.attributes) {
+      if (!(key.startsWith('_') || excluded_keys.includes(key))) {
+        this.listenTo(this.model, 'change:' + key, () => {
+          this.setControlOptions();
+
+          this.map_view.obj.pm.removeControls();
+          if (!this.model.get('hide_controls')) {
+            this.map_view.obj.pm.addControls(this.controlOptions);
+          }
```

Comment:
This part of code is also being repeated in `setPosition()` below, maybe you want to take it out in a separate method of in you have any better idea?